### DISCUSSION TITLE
bare strings

### DISCUSSION THREAD

**Discussion by hclarke** [kdl-org/kdl#48] [Ideas]
The idea here would be to allow bare strings, using the identifier syntax

```diff
- value := string | raw_string | number | boolean | 'null'
+ value := raw_string | number | boolean | 'null' | identifier
```

and then you could have these parse as the same thing:
```
foo bar
foo "bar"
```

thoughts?

**Comment by zkat** [comment]
One thing I like about KDL when compared to YAML in particular is that types are explicit and unambiguous. That you don't generally have to think about parsing rules in order to get something right, and typos are relatively easy to find and correct.

I feel like doing these sorts of strings might introduce such ambiguity. Aside from the possible confusion around numbers or bool or null, I see the potential for some strange results if you decide to put spaces in your unquoted strings. Because they'll be treated as multiple values, even though a naive reader might think they're dealing with regular key value pairs.

**Comment by eugenesvk** [comment]
That would be a great addition to KDL as I think you should only quote to resolve ambiguity instead of requiring the verbosity of `"` everywhere. This would be similar to how node names can have a lot of symbols without resorting to quotes (an awesome design feature of KDL!), but then to add a few more symbols that are not allowed, you need `"`

I understand that it's potentially more confusing with strings (though wouldn't raise to the level of the infamous old `No`rway problem in Yaml), but that's partially solved via using proper tools like syntax highlighting, and still think it's a worthy tradeoff, e.g., for the cleanest config file experience ({"hello":"json"})
